homeassistant/components/sensor/alpha_vantage.py

```diff
@@ -162,6 +168,8 @@ def update(self):
         all_values, _ = self._timeseries.get_intraday(self._symbol)
         self.values = next(iter(all_values.values()))
         _LOGGER.debug("Received new values for symbol %s", self._symbol)
+        _LOGGER.debug("Sleep %s seconds to prevent API throttling...", 10)
+        time.sleep(10)
```

Comment:
Instead of doing I/O on `setup_platforms` it should be done inside `update`. Instead of sleeping you should use `homeassistant.util.Throttle` decorator (grep for `Throttle` on the codebase to see examples).

Reply:
Well. It (also) needs to be done in some way during setup of the platforms. The API is being hit in the `setup_platforms` **as well as** in the `update`.

Can a `Throttle` be used for `setup_platforms`, too?

I will look into the Throttle for the update and add a commit.

Reply:
`setup_platform` is only called once when the platform is initialized, so there's no need to throttle that.

Reply:
Well, that's exactly my point. The reason I started thinking about a PR in the first place was the platform setup. If you have more that a few ( 1 or 2) symbols, then the current platform setup will show errors because of throttling. The setup will call the API for each symbol. See line 90: 

    timeseries.get_intraday(symbol[CONF_SYMBOL])

5, 6 or 10 of these calls and the API will complain (and those sensors will not be created at all).